src/ansi-c/c_typecheck_code.cpp

```diff
@@ -842,35 +843,158 @@ void c_typecheck_baset::typecheck_dowhile(code_dowhilet &code)
   }
 }
 
+void c_typecheck_baset::typecheck_spec_assigns_condition(exprt &condition)
+{
+  // compute type
+  typecheck_expr(condition);
+
+  // non-fatal warnings
+  if(has_subexpr(condition, [&](const exprt &subexpr) {
+       return can_cast_expr<side_effect_expr_function_callt>(subexpr);
+     }))
```

Comment:
I'm not entirely sure, but the following might also work (and would be shorter):
`if(has_subexpr(condition, can_cast_expr<side_effect_expr_function_callt>))`